Samples/Widgets/MultipleWidgets/cpp-win32-packaged/SampleWidgetProviderApp/main.cpp

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "WidgetProvider.h"
+#include <mutex>
+
+// This GUID is the same GUID that was provided in the 
+// registration of the COM Server and Class Id in the .appxmanifest.
+static constexpr GUID widget_provider_clsid
+{ /* 101D03A3-6FC8-4887-9B64-310A7B164319 */
+    0x101d03a3, 0x6fc8, 0x4887, {0x9b, 0x64, 0x31, 0x0a, 0x7b, 0x16, 0x43, 0x19}
+};
+
+// This is implementation of a ClassFactory that will instantiate WidgetProvider
+// that will interact with the Widget Service.
+template <typename T>
+struct SingletonClassFactory : winrt::implements<SingletonClassFactory<T>, IClassFactory>
+{
```

Comment:
@beervoley Was there any discussion about IClassFactory usage here? Feels like we're mixing classic COM and WinRT here, which I suspect will greatly impact C# and other languages. (Not sure C#/WinRT supports this, for example.) Now may be a great time to break this ABI and switch to IActivationFactory before widgets goes GA and it's too late...

@jonwis

Reply:
@ChrisGuzak - could you please advise here?:)

Reply:
@riverar usage of IActivationFactory requires usage of `RoRegisterActivationFactories`. We cannot register a factory for `WinRT` object `system wide` so they can be activated OOP `outside of the registering app package graph`. This is a unfortunate limitation, but it is by design.
COM objects in a package do have `system scope` and they don't have this issue :)

To my best knowledge, we can replicate this in C# .NET :)